codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/route53/Route53TrimHostedZone.kt

```diff
@@ -38,12 +39,11 @@ class Route53TrimHostedZone : SwiftIntegration {
         val inputShape = MiddlewareShapeUtils.inputShape(ctx.model, operationShape)
         val hostedZoneMember = inputShape.members().find { it.hasTrait<TrimHostedZone>() }
         if (hostedZoneMember != null) {
-            StripHostedZoneURLPathMiddleware.renderMiddleware(ctx, operationShape)
-            operationMiddleware.prependMiddleware(operationShape, StripHostedZoneUrlPathMiddlewareRenderable(ctx.model, ctx.symbolProvider))
+            operationMiddleware.prependMiddleware(operationShape, TrimHostedZoneURLPathMiddlewareRenderable(ctx.model, ctx.symbolProvider))
         }
     }
 
     private fun isHostId(shape: Shape): Boolean {
-        return (shape is MemberShape && shape.target == ShapeId.from("com.amazonaws.route53#ResourceId")) && shape.hasTrait<HttpLabelTrait>() && shape.memberName == "HostedZoneId"
+        return (shape is MemberShape && shape.target == ShapeId.from("com.amazonaws.route53#ResourceId")) && shape.hasTrait<HttpLabelTrait>()
```

Comment:
The overly-restrictive requirement that the member be named `HostedZoneId` is removed.